geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/ingest/ConverterInputFormat.scala

```diff
@@ -0,0 +1,132 @@
+/***********************************************************************
+* Copyright (c) 2013-2015 Commonwealth Computer Research, Inc.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Apache License, Version 2.0 which
+* accompanies this distribution and is available at
+* http://www.opensource.org/licenses/apache2.0.php.
+*************************************************************************/
+package org.locationtech.geomesa.tools.ingest
+
+import com.typesafe.config.ConfigFactory
+import com.typesafe.scalalogging.LazyLogging
+import org.apache.commons.io.IOUtils
+import org.apache.hadoop.fs.{FileSystem, Path}
+import org.apache.hadoop.io.LongWritable
+import org.apache.hadoop.io.compress.{CodecPool, Decompressor, CompressionCodecFactory}
+import org.apache.hadoop.mapreduce._
+import org.apache.hadoop.mapreduce.lib.input.{FileInputFormat, FileSplit}
+import org.locationtech.geomesa.convert.Transformers.EvaluationContext
+import org.locationtech.geomesa.convert.{SimpleFeatureConverter, SimpleFeatureConverters}
+import org.locationtech.geomesa.tools.ingest.ConverterInputFormat._
+import org.locationtech.geomesa.utils.geotools.SimpleFeatureTypes
+import org.opengis.feature.simple.{SimpleFeature, SimpleFeatureType}
+
+/**
+ * Input format for Converters gives us access to the entire file as a byte stream
+ * via the record reader. In the future we can use this to implement the processing
+ * of InputStreams from the Converter API so that we don't have to load the entire
+ * file into memory (esp for multi-line xml, json, avro).
+ */
+class ConverterInputFormat extends FileInputFormat[LongWritable, SimpleFeature] with LazyLogging {
+  override protected def isSplitable(context: JobContext, filename: Path): Boolean = false
+
+  override def createRecordReader(split: InputSplit, context: TaskAttemptContext): RecordReader[LongWritable, SimpleFeature] = {
+    new ConverterRecordReader()
+  }
+}
+
+object ConverterInputFormat {
+
+  object Counters {
+    val Group  = "org.locationtech.geomesa.jobs.convert"
+    val Success = "success"
+    val Failure = "failure"
+    val Written = "written"
+    val LineCount = "lineCount"
+  }
+
+  val ConverterKey  = "org.locationtech.geomesa.jobs.ingest.converter"
+  val SftKey        = "org.locationtech.geomesa.jobs.ingest.sft"
+  val TypeNameKey   = "org.locationtech.geomesa.jobs.ingest.sft.name"
```

Comment:
yeah...the scala folks claim Camel Case...if we have a real preference for UPP_STUFF I'll revert to that :)